### PR TITLE
Remove `disable_vectors` legacy KB setting

### PR DESCRIPTION
**Pull request by lferran**
### Description
This setting does not seem to be used anymore.
It has polluted the process of serializing search results by introducing an extra maindb lookup for the KB config, simply to get the value of `disable_vectors`.

### How was this PR tested?
Existing tests should cover us

### CODE REVIEW

**Review by tarekziade — APPROVED** [review]

**Comment by lferran** [comment]
waiting on @bloodbare 's approval to merge.

**Comment by lferran** [comment]
> waiting on @bloodbare 's approval to merge.

he approved verbally to me!

**Pull request merged**
